src/timesync/timesyncd-bus.c

```diff
@@ -43,6 +49,58 @@ static int property_get_servers(
         return sd_bus_message_close_container(reply);
 }
 
+static int method_set_runtime_servers(sd_bus_message *message, void *userdata, sd_bus_error *error) {
+        _cleanup_strv_free_ char **msg_names = NULL;
+        Manager *m = userdata;
+        int r;
+
+        assert(m);
+        assert(message);
+
+        r = sd_bus_message_read_strv(message, &msg_names);
+        if (r < 0)
+                return r;
+
+        STRV_FOREACH(name, msg_names) {
+                r = dns_name_is_valid_or_address(*name);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to check validity of NTP server name or address '%s': %m", *name);
+                if (r == 0)
+                        return sd_bus_error_setf(error, SD_BUS_ERROR_INVALID_ARGS, "Invalid NTP server name or address, refusing: %s", *name);
+        }
+
+        r = bus_verify_polkit_async(message, CAP_NET_ADMIN,
+                                    "org.freedesktop.timesync1.set-runtime-servers",
+                                    NULL, true, UID_INVALID,
+                                    &m->polkit_registry, error);
+        if (r < 0)
+                return r;
+        if (r == 0)
+                /* Polkit will call us back */
+                return 1;
+
+        manager_flush_runtime_servers(m);
```

Comment:
`manager_flush_runtime_servers()` clears previous runtime servers, and if current server is a runtime server, then disconnect from the server.

However, it does not disconnect when a static system server or link server is currently used.
I think, it is better to call `manager_set_server_name(m, NULL)` before `manager_connect()`. Then, the specified new runtime server is used.

Reply:
Okay, this is a change in behavior but I'm happy with that. My intention was to not disconnect from currently connected servers but only to memorize runtime servers for cases when there is no active connection.

But it doesn't matter in my case, and probably not in anyone else's either. After all, the whole point of introducing the runtime servers is to support systems where all other means of providing that information fails, which means that all other lists are empty anyway.

Changed now.